addons/sale_product_configurator/controllers/main.py

```diff
@@ -61,7 +61,8 @@ def _show_optional_products(self, product_id, variant_values, pricelist, handle_
         combination = request.env['product.template.attribute.value'].browse(variant_values)
         has_optional_products = product.optional_product_ids.filtered(lambda p: p._is_add_to_cart_possible(combination))
 
-        if not has_optional_products:
+        if not has_optional_products and (product.product_variant_count <= 1 or variant_values):
+            # The modal is not shown if there are no optional products and the main product either has no variants or is already configured
```

Comment:
I don't really understand the change here...
Why would we show the optional products if there are no optional products ? And if we want to do so, the method name should be updated :thinking:.

Reply:
> Why would we show the optional products if there are no optional products

As you spotted, the name is not more exact...

But we open it to allow user to customize the product added in his cart from the /shop page...
I didnt check the PR and how it has been done... You are too fast for me...

But goal of this task is when you add a Customize Desk on /shop page, before we add for you the first variant of the product.template shown... now, we propose you to customize it and chose which variant you want.

How to test the functionality: add customize desk from /shop page before / after this pr

Reply:
@Feyensv I will try to find a better name for the functionality, it's indeed not very clear. 

**The added configuration step is needed for some cases like the following:**

1. When adding a product that had variants directly from the /shop page before, the first variant was added by default and configuring it was impossible without explicitly going on the product's page and configuring it from there. This behaviour made no sense.
2. When going through the product's page, and you have optional products enabled for the product you are configuring, then it's *nice* but not necessary to be able to configure the product further while choosing and configuring the optional products. This doesn't add an extra step in any case, it just makes it possible to still configure your product in the modal.
3. If there are no optional products, the configuration you have made on the product's page is taken into account and the product is directly added to the cart without opening the modal, which is the same behaviour as previously.

The bug you mention where the values you've entered are not kept when you close the modal is something that was already the case before the change, and I can look into fixing it and adding the fix to this PR if deemed necessary 😃

Reply:
Was this taken into account?
> Why would we show the optional products if there are no optional products ? **And if we want to do so, the method name should be updated .**